build_tools/circle/build_doc.sh

```diff
@@ -51,5 +52,11 @@ source activate testenv
 # Build and install scikit-learn in dev mode
 python setup.py develop
 
+if [[ "$CIRCLE_BRANCH" =~ ^master$|^[0-9]+\.[0-9]+\.X$ ]]
```

Comment:
Actually thinking about it, I remember now that CircleCI builds in branches and not (as you kind of would expect from Travis) on the result of the merge of the PR into master, so CIRCLE_BRANCH may be the name of the branch in the fork.

Still it's better to make sure that we never build the pdf in a PR, for example some beginners create PRs from their master branch sometimes.